### ISSUE TITLE
SELECT WITH where inner query has columns with same name ignores table aliases

### ISSUE THREAD

**Issue by gsora** [sqlc-dev/sqlc#1236]
### Version

1.10.0

### What happened?

Given the following query:

```sql
WITH
	q
		AS (
			SELECT
				authors.name, authors.bio
			FROM
				authors
				LEFT JOIN fake ON authors.name = fake.name
		)
SELECT
	*
FROM
	q AS c1
WHERE
-- the following conditions have issues:
c1.name = $1;
-- this condition doesn't have issues:
-- c1.name = 'lol';
-- nor this one:
-- authors.name = $1;
```

`sqlc generate` fails to generate a query, because the `name` specifier is ambiguous.

### Relevant log output

```shell
query.sql:29:1: column reference "name" is ambiguous
```

### Database schema

```sql
CREATE TABLE authors (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text
);

CREATE TABLE fake (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text
);
```

### SQL queries

```sql
WITH
	q
		AS (
			SELECT
				authors.name, authors.bio
			FROM
				authors
				LEFT JOIN fake ON authors.name = fake.name
		)
SELECT
	*
FROM
	q AS c1
WHERE
-- the following conditions have issues:
c1.name = $1;
-- this condition doesn't have issues:
-- c1.name = 'lol';
-- nor this one:
-- authors.name = $1;
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/31d37017dbab42342784faecfab1f0b4a76b48bce3dbd5b5cb433fea68d878ea

### What operating system are you using?

Linux

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by Davincible** [comment]
Same here

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/cte_left_join/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/36b5a02a1d223a6bb032892410bbb52c8665895c3b4482ceb79f30846ad0a344)
